### ISSUE TITLE
[Bug/RFE] oscap ds sds-compose does not recognize OCIL file as SCAP-1.2 check system and places it into <ds:extended-components> element (intended for non-SCAP content) rather into <ds:checks> element (intended for SCAP-1.2 check systems)

### ISSUE THREAD

**Issue by iankko** [OpenSCAP/openscap#364]
```oscap``` currently (as of 1.2.8) version does not support / recognize OCIL files. Snippet from SSG ```RHEL/6``` product build (providing only relevant parts):
```
$ pwd
/tmp/github/scap-security-guide/RHEL/6
$ make content
..
oscap ds sds-compose output/ssg-rhel6-xccdf-1.2.xml output/ssg-rhel6-ds.xml
OpenSCAP Error: Unknown document type: 'ocil-ssg.xml' [oscap_source.c:172]
```

This is not an issue (I realize adding OCIL support to OpenSCAP might not be trivial task).
But the issue is that instead of inserting the ```<ocil:ocil>``` component into ```<ds:checks>``` datastream element (as required by [SCAP 1.2 standard](http://csrc.nist.gov/publications/nistpubs/800-126-rev2/SP800-126r2.pdf)), the ```<ocil:ocil>``` element is placed into ```<ds:extended-components>``` element (intended for non-SCAP content) when creating a datastream.

This results into issues like, e.g.:
* https://github.com/OpenSCAP/scap-security-guide/issues/1100
* https://github.com/OpenSCAP/scap-security-guide/issues/1101

when trying to validate such a created datastream file with e.g. SCAP NIST test content suite.

Reproducible:
Always

Steps to reproduce:
* Build datastream e.g. for ```RHEL/6``` SSG product (```ssg-rhel6-ds.xml```) by issuing ```make content``` in ```RHEL/6``` folder,
* Validate such datastream with SCAP NIST content validation testsuite

Current result (the relevant parts of the produced datastream look like) [Were equipped with newlines for better reading]:
```
<?xml version="1.0" encoding="utf-8"?>
<ds:data-stream-collection xmlns:ds="http://scap.nist.gov/schema/scap/source/1.2" 
xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:cat="urn:oasis:names:tc:entity:xmlns:xml:catalog" 
id="scap_org.open-scap_collection_from_xccdf_ssg-rhel6-xccdf-1.2.xml" schematron-version="1.0">

<ds:data-stream id="scap_org.open-scap_datastream_from_xccdf_ssg-rhel6-xccdf-1.2.xml" 
scap-version="1.2" use-case="OTHER">

  <ds:dictionaries>
    <ds:component-ref id="scap_org.open-scap_cref_output--ssg-rhel6-cpe-dictionary.xml" 
xlink:href="#scap_org.open-scap_comp_output--ssg-rhel6-cpe-dictionary.xml">
      <cat:catalog>
        <cat:uri name="ssg-rhel6-cpe-oval.xml" uri="#scap_org.open-scap_cref_output--ssg-rhel6-
cpe-oval.xml"/>
      </cat:catalog>
    </ds:component-ref>
  </ds:dictionaries>

  <ds:checklists>
    <ds:component-ref id="scap_org.open-scap_cref_ssg-rhel6-xccdf-1.2.xml" 
xlink:href="#scap_org.open-scap_comp_ssg-rhel6-xccdf-1.2.xml">
      <cat:catalog>
        <cat:uri name="ssg-rhel6-oval.xml" uri="#scap_org.open-scap_cref_ssg-rhel6-oval.xml"/>
        <cat:uri name="ocil-ssg.xml" uri="#scap_org.open-scap_cref_ocil-ssg.xml"/>
      </cat:catalog>
    </ds:component-ref>
  </ds:checklists>

  <ds:checks>
    <ds:component-ref id="scap_org.open-scap_cref_ssg-rhel6-oval.xml" xlink:href="#scap_org.open-
scap_comp_ssg-rhel6-oval.xml"/>
    <ds:component-ref id="scap_org.open-scap_cref_output--ssg-rhel6-cpe-oval.xml" 
xlink:href="#scap_org.open-scap_comp_output--ssg-rhel6-cpe-oval.xml"/>
    <ds:component-ref id="scap_org.open-scap_cref_output--ssg-rhel6-oval.xml" 
xlink:href="#scap_org.open-scap_comp_output--ssg-rhel6-oval.xml"/>
  </ds:checks>

  <ds:extended-components>
    <ds:component-ref id="scap_org.open-scap_cref_ocil-ssg.xml" xlink:href="#scap_org.open-
scap_ecomp_ocil-ssg.xml"/>
  </ds:extended-components>

</ds:data-stream>

..
</oval_definitions></ds:component><ds:extended-component id="scap_org.open-scap_ecomp_ocil-
ssg.xml" timestamp="2016-03-25T10:24:15"><ocil xmlns:xsi="http://www.w3.org/2001/XMLSchema-
instance" xmlns="http://scap.nist.gov/schema/ocil/2.0" xmlns:xhtml="http://www.w3.org/1999/xhtml">
  <generator>
    <schema_version>2.0</schema_version>
    <timestamp>2016-03-25T10:24:10-04:00</timestamp>
  </generator>
  <questionnaires>
...
   </boolean_question>
  </questions>
</ocil></ds:extended-component></ds:data-stream-collection>
```

See also attached ```invalid-ssg-rhel6-ds.xml``` for a complete example.

Expected result (again only relevant parts separated by newlines for better readability):
```
<?xml version="1.0" encoding="utf-8"?>
<ds:data-stream-collection xmlns:ds="http://scap.nist.gov/schema/scap/source/1.2" 
xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:cat="urn:oasis:names:tc:entity:xmlns:xml:catalog" 
id="scap_org.open-scap_collection_from_xccdf_ssg-rhel6-xccdf-1.2.xml" schematron-version="1.0">

<ds:data-stream id="scap_org.open-scap_datastream_from_xccdf_ssg-rhel6-xccdf-1.2.xml" 
scap-version="1.2" use-case="OTHER">

  <ds:dictionaries>
    <ds:component-ref id="scap_org.open-scap_cref_output--ssg-rhel6-cpe-dictionary.xml" 
xlink:href="#scap_org.open-scap_comp_output--ssg-rhel6-cpe-dictionary.xml">
      <cat:catalog>
        <cat:uri name="ssg-rhel6-cpe-oval.xml" uri="#scap_org.open-scap_cref_output--ssg-rhel6-
cpe-oval.xml"/>
      </cat:catalog>
    </ds:component-ref>
  </ds:dictionaries>

  <ds:checklists>
    <ds:component-ref id="scap_org.open-scap_cref_ssg-rhel6-xccdf-1.2.xml" 
xlink:href="#scap_org.open-scap_comp_ssg-rhel6-xccdf-1.2.xml">
      <cat:catalog>
        <cat:uri name="ssg-rhel6-oval.xml" uri="#scap_org.open-scap_cref_ssg-rhel6-oval.xml"/>
        <cat:uri name="ocil-ssg.xml" uri="#scap_org.open-scap_cref_ocil-ssg.xml"/>
      </cat:catalog>
    </ds:component-ref>
  </ds:checklists>

  <ds:checks>
    <ds:component-ref id="scap_org.open-scap_cref_ssg-rhel6-oval.xml" xlink:href="#scap_org.open-
scap_comp_ssg-rhel6-oval.xml"/>
    <ds:component-ref id="scap_org.open-scap_cref_output--ssg-rhel6-cpe-oval.xml" 
xlink:href="#scap_org.open-scap_comp_output--ssg-rhel6-cpe-oval.xml"/>
    <ds:component-ref id="scap_org.open-scap_cref_output--ssg-rhel6-oval.xml" 
xlink:href="#scap_org.open-scap_comp_output--ssg-rhel6-oval.xml"/>
    <ds:component-ref id="scap_org.open-scap_cref_ocil-ssg.xml" xlink:href="#scap_org.open-
scap_comp_ocil-ssg.xml"/>
  </ds:checks>

</ds:data-stream>
..
<ds:component id="scap_org.open-scap_comp_ocil-ssg.xml" timestamp="2016-03-25T10:24:15">
<ocil xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://scap.nist.gov/schema
/ocil/2.0" xmlns:xhtml="http://www.w3.org/1999/xhtml">
  <generator>
    <schema_version>2.0</schema_version>
    <timestamp>2016-03-25T10:24:10-04:00</timestamp>
..
                        Is it the case that ?
                        </question_text>
    </boolean_question>
  </questions>
</ocil>
</ds:component>
</ds:data-stream-collection>
```

In other words:
* the ```<ocil:ocil>``` datastream component should be part of ```<ds:checks>``` datastream element rather than ```<ds:extended-components>``` element,
* the ```<ds:extended-component>``` tag for OCIL component should be replaced by ```<ds:component>``` tag

See also ```valid-ssg-rhel6-ds.xml``` for a complete datastream example.

Additional information:
See ```Table 3 - ds:data-stream```, ```Table 6 - ds:checks```, and section ```3.2.4.2 The <xccdf:check> Element``` for information on OCIL check system support and requirements in [SCAP v1.2](http://csrc.nist.gov/publications/nistpubs/800-126-rev2/SP800-126r2.pdf) standard.

**Comment by iankko** [comment]
Previously promised ```invalid-ssg-rhel6-ds.xml``` datastream file (as currently produced by oscap):

[invalid-ssg-rhel6-ds.xml.zip](https://github.com/OpenSCAP/openscap/files/190047/invalid-ssg-rhel6-ds.xml.zip)

And ```valid-ssg-rhel6-ds.xml``` datastream file with OCIL component as it should look like after the SCAP 1.2 standard requirements:

[valid-ssg-rhel6-ds.xml.zip](https://github.com/OpenSCAP/openscap/files/190051/valid-ssg-rhel6-ds.xml.zip)

**Comment by jan-cerny** [comment]
Current OpenSCAP from git maint-1.2 HEAD even thinks that the OCIL is a SCE script.
It seems like it was caused by implementing support for SCE in datastreams, which was done recently (after 1.2.8 release).

A small snippet from output of `oscap ds sds-compose` follows:

````
.....
</Benchmark></ds:component><ds:extended-component xmlns:oscap-sce-xccdf-stream="http://open-scap.org/pag
e/SCE_xccdf_stream" id="scap_org.open-scap_ecomp_ssg-rhel6-ocil.xml" timestamp="2016-04-22T09:45:05"><os
cap-sce-xccdf-stream:script>&lt;ocil xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http:/
/scap.nist.gov/schema/ocil/2.0" xmlns:xhtml="http://www.w3.org/1999/xhtml"&gt;
  &lt;generator&gt;
    &lt;schema_version&gt;2.0&lt;/schema_version&gt;
    &lt;timestamp&gt;2016-04-22T09:45:01+02:00&lt;/timestamp&gt;
  &lt;/generator&gt;
  &lt;questionnaires&gt;
    &lt;questionnaire id="ocil:ssg-partition_for_tmp_ocil:questionnaire:1"&gt;
      &lt;title&gt;Ensure /tmp Located On Separate Partition&lt;/title&gt;
......
```

**Comment by jan-cerny** [comment]
Fixed in #398
